tests/min-version.test.ts

```diff
@@ -0,0 +1,57 @@
+import { expect, test } from '@playwright/test';
+import { gotoHtmlWidget } from './utlis/server/playground';
+import { prepareNetwork, printlnCode } from './utlis';
+import { mockRunRequest, waitRunRequest } from './utlis/mocks/compiler';
+import { runButton } from './utlis/interactions';
+import {
+  LOADER_SELECTOR,
+  RESULT_SELECTOR,
+  VERSION_SELECTOR,
+  WIDGET_SELECTOR,
+} from './utlis/selectors';
+
+test.describe('Minimum compiler version', () => {
+  test.beforeEach(async ({ page, baseURL }) => {
+    await prepareNetwork(page, baseURL); // offline mode
+  });
+
+  test('with future release', async ({ page }) => {
+    await gotoHtmlWidget(
+      page,
+      { selector: 'code' },
+      /* language=html */ `
+        <code data-min-compiler-version='2.0.0'>${printlnCode(
+          'Hello, world!',
+        )}</code>
+      `,
+    );
+
+    const editor = page.locator(WIDGET_SELECTOR);
+    await expect(editor).toHaveCount(1); // playground loaded
+    await expect(page.locator('code')).not.toBeVisible(); // original node hided
+    await expect(editor.locator(VERSION_SELECTOR)).toHaveText(
+      'Running on v.2.0.0-alpha.3', // latest version marker
+    );
```

Comment:
It would be great to encapsulate this logic into a function to add an abstraction level.
Now, too low-level manipulations go there, and I assume they duplicate each other across the tests.
For example:
```
await expect(editor).toHaveCount(1);
```
Might be something like:
```
await utils.isPlaygroundLoaded();
```
It would help to make this a more readable code (you were forced to add a comment to explain what happens here) and provide an easy way to update the code if the implementation changes (instead of one place change, now you need to update all the places in code where such a line exists).